### PR TITLE
Add WeakSet edge case tests

### PR DESCRIPTION
**Pull request by mrhapile**
### This PR expands the WeakSet unit test coverage.

_The initial WeakSet tests introduced basic behavioral tests._
**#4912** 
This PR adds additional ECMAScript edge case coverage.

The following behaviors are now tested:

- Primitive rejection in WeakSet.prototype.add
- Primitive-safe behavior for WeakSet.prototype.has
- Primitive-safe behavior for WeakSet.prototype.delete
- Duplicate object additions

These tests ensure correct handling of **ECMAScript §24.4 WeakSet**
invariants and improve robustness during future refactors.

Screenshot 2026-03-07 at 7 15 01 AM

All tests follow the same pattern used by other builtin tests.

### CODE REVIEW

File: core/engine/src/builtins/weak_set/tests.rs [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+        ),
+        TestAction::assert_native_error(
+            "ws.add(true);",
+            JsNativeErrorKind::Type,
+            "WeakSet.add: expected target argument of type `object`, got target of type `boolean`",
+        ),
+        TestAction::assert_native_error(
+            "ws.add(null);",
+            JsNativeErrorKind::Type,
+            "WeakSet.add: expected target argument of type `object`, got target of type `object`",
```
[the remark this answers is not in the document]
Comment (jedel1043) [reply core/engine/src/builtins/weak_set/tests.rs:51]: it's fine, we'll fix it eventually

**Review by jedel1043 — APPROVED** [review]
Looks good

File: core/engine/src/builtins/weak_set/tests.rs (hunk 1 above)
[the remark this answers is not in the document]
Comment (mrhapile) [reply core/engine/src/builtins/weak_set/tests.rs:51]: rn I was working on this fix only should I push it 🙆‍♂️!?
